Yotpo/Yotpo/Controller/Adminhtml/YotpoController/YotpoController.php

```diff
@@ -0,0 +1,116 @@
+<?php
+
+namespace Yotpo\Yotpo\Controller\Adminhtml\YotpoController;
+ 
+class YotpoController extends \Magento\Backend\App\Action
+{
+
+//max amount of orders to export
+const MAX_ORDERS_TO_EXPORT = 5000;
+const MAX_BULK_SIZE        = 200;
+
+protected $_messageManager;
+
+public function __construct(
+        \Magento\Framework\App\Action\Context $context,
+        \Magento\Framework\App\Request\Http $request,
+        \Magento\Framework\App\Response\Http $response,
+        \Magento\Store\Model\StoreManagerInterface $storeManager,
+        \Yotpo\Yotpo\Block\Config $config,
+        \Yotpo\Yotpo\Helper\ApiClient $api,
+        \Magento\Framework\Message\ManagerInterface $messageManager,
+        \Magento\Framework\Controller\Result\RedirectFactory $resultRedirectFactory,
+        \Psr\Log\LoggerInterface $logger
+    ) {
+        $this->_request = $request;
+        $this->_response = $response; 
+        $this->_storeManager = $storeManager;
+        $this->_config = $config;
+        $this->_api = $api;
+        $this->_logger = $logger;
+        $this->_messageManager = $messageManager;          
+        $this->_resultRedirectFactory = $resultRedirectFactory;   
+        parent::__construct($context);
+    }
+
+
+    /**
+     * say admin text
+     */
+    public function execute()
```

Comment:
The controller's execute function is called when the button is clicked and then an ajax request calls the controller's execute